### ISSUE TITLE
docker-machine provisions docker to use aufs on debian jessie which is not supported

### ISSUE THREAD

**Issue by raarts** [docker/machine#3895]
docker-machine version 0.8.2, build e18a919
host machine: Debian Jessie 8.4

command used: `docker-machine create -d generic --generic-ip-address=x.x.x.x --generic-ssh-key ~/.ssh/id_rsa  node-1`

This results in `Error starting daemon: error initializing graphdriver: driver not supported` and docker-machine failing.

**Comment by nathanleclaire** [comment]
hmm... odd... jessie should support the AUFS patch just fine...

**Comment by nathanleclaire** [comment]
can you post the full `docker-machine --debug` output?

**Comment by ldumont** [comment]
I had the same kind of trouble last year when I tried to do the same, but I gave it another try this week and I was able to make it work.

docker-machine version 0.8.2 
Jessie 8.6 kernel 3.16 => straightforward setup
Jessie 8.6 kernel 4.8.11 from backports => need to add --engine-storage-driver overlay otherwise aufs is used and it's not supported in recent kernels

I don't know if something change since last year but I was not able to make it work, so give it another try !

**Comment by raarts** [comment]
Yes I was able to make it work too, by specifying  --engine-storage-driver
overlay, but I t hink that should be detected and automated in some way.
Currently it works with some cloud providers and not with others.

**Comment by dminkovsky** [comment]
Related #4016 #4156

**Comment by jak** [comment]
Creating machines on Google Cloud platform currently fails due to this.

**Comment by mgmorcos** [comment]
+1

**Comment by luckydonald** [comment]
Is it somehow possible to manually set something else?
Machine creates one with `aufs`, which is the platform [default](https://github.com/docker/machine/blob/97c1136d1ec9ae4c3ab69fda615e3dd577809e5c/libmachine/provision/debian.go#L93)


`/etc/systemd/system/docker.service.d/10-machine.conf`
```systemd
[Service]
ExecStart=
ExecStart=/usr/bin/dockerd -H tcp://0.0.0.0:2376 -H unix:///var/run/docker.sock --storage-driver aufs --tlsverify --tlscacert /etc/docker/ca.pem --tlscert /etc/docker/server.pem --tlskey /etc/docker/server-key.pem --label provider=generic
Environment=
```
